### PR TITLE
Added CBMC proofs for OTA_MQTT_strerror function

### PR DESCRIPTION
**Pull request by Bujain**
*Description of changes:*
This PR contains the CBMC proofs for the OTA_MQTT_strerror function. The files added to this PR are:

OTA_MQTT_strerror/Makefile
OTA_MQTT_strerror/cbmc-proof.txt
OTA_MQTT_strerror/cbmc-viewer.json
OTA_MQTT_strerror/OTA_MQTT_strerror_harness.c
OTA_MQTT_strerror/README.md

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/Makefile-template-defines [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # Absolute path to the root of the source tree.
 #
 SRCDIR ?= $(abspath $(PROOF_ROOT)/../../../)
-
+>
```
Comment (yanjos-dev) [inline test/cbmc/proofs/Makefile-template-defines:5]: What is this character? Was this added by accident?

File: test/cbmc/proofs/OTA_MQTT_strerror/OTA_MQTT_strerror_harness.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+void OTA_MQTT_strerror_harness()
+{
+    OtaMqttStatus_t status;
+    const char * pErrorMsg;
+
+    pErrorMsg = OTA_MQTT_strerror( status );
+
+    /* The function returns a string which is never NULL. If it does, then it is a
+     *  problem. */
```
Comment (yanjos-dev) [inline test/cbmc/proofs/OTA_MQTT_strerror/OTA_MQTT_strerror_harness.c:38]: It's true that based on the implementation, the string can not be returned as a non-NULL value. Kind of a nitpick, but the assertion is more about the expectation of the caller and less about the implementation. The assert is to confirm that the implementation is doing what we expect it to be doing. So I would reword it to be something like this:
```suggestion
    /* The caller of this function assumes that the return value is never NULL. */
```

File: test/cbmc/proofs/OTA_MQTT_strerror/OTA_MQTT_strerror_harness.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    __CPROVER_assert( pErrorMsg != NULL, "Function return cannot be NULL" );
```
Comment (yanjos-dev) [inline test/cbmc/proofs/OTA_MQTT_strerror/OTA_MQTT_strerror_harness.c:39]: Please make the error string a full sentence that ends in a period. Also, this error string could be more descriptive about what actually happened. "Function return cannot be NULL" is stating an assumption, not an error.

**Review by yanjos-dev — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/Makefile-template-defines (hunk 1 above)
Comment (Bujain) [reply test/cbmc/proofs/Makefile-template-defines]: Removed it in 207cf11

File: test/cbmc/proofs/OTA_MQTT_strerror/OTA_MQTT_strerror_harness.c (hunk 2 above)
Comment (Bujain) [reply test/cbmc/proofs/OTA_MQTT_strerror/OTA_MQTT_strerror_harness.c]: Updated it in 207cf11

File: test/cbmc/proofs/OTA_MQTT_strerror/OTA_MQTT_strerror_harness.c (hunk 3 above)
Comment (Bujain) [reply test/cbmc/proofs/OTA_MQTT_strerror/OTA_MQTT_strerror_harness.c]: Updated it in 207cf11

**Review by yanjos-dev — APPROVED** [review]

**Review by aggarg — APPROVED** [review]
